app/models/publisher.rb

```diff
@@ -164,6 +164,26 @@ def set_uphold_updated_at
     self.uphold_updated_at = Time.now
   end
 
+  def uphold_scope
+    wallet.try(:wallet_details).try(:[], 'scope')
+  end
+
+  def uphold_authorized?
+    wallet.try(:wallet_details).try(:[], 'authorized')
+  end
+
+  def eyeshade_default_currency
+    wallet.try(:wallet_details).try(:[], 'defaultCurrency')
+  end
+
+  def available_uphold_currencies
+    wallet.try(:wallet_details).try(:[], 'availableCurrencies')
+  end
+
+  def possible_uphold_currencies
+    wallet.try(:wallet_details).try(:[], 'possibleCurrencies')
+  end
+
```

Comment:
I'd like to move these methods to the wallet class. It would make it cleaner to remove the wallet from the publisher if we choose to move to a caching scheme. That could help clean up the flow when wallet modifying actions (like the `CardCreationService` and `PublisherDefaultCurrencySetter`) are taken in the same request (we could invalidate the cache afterwards and then the next action would pull a fresh wallet).

Reply:
I will move these to the wallet class.